shared/actions/deeplinks.tsx

```diff
@@ -0,0 +1,113 @@
+import * as ChatGen from './chat2-gen'
+import * as Constants from '../constants/config'
+import * as Container from '../util/container'
+import * as DeeplinksGen from './deeplinks-gen'
+import * as ProfileGen from './profile-gen'
+import * as RouteTreeGen from './route-tree-gen'
+import * as Saga from '../util/saga'
+import * as Tabs from '../constants/tabs'
+import * as WalletsGen from './wallets-gen'
+import URL from 'url-parse'
+
+const handleKeybaseLink = (_: Container.TypedState, action: DeeplinksGen.HandleKeybaseLinkPayload) => {
+  const error =
+    "We couldn't read this link. The link might be bad, or your Keybase app might be out of date and need to be updated."
+  const parts = action.payload.link.split('/')
+  // List guaranteed to contain at least one elem.
+  switch (parts[0]) {
+    case 'profile':
+      if (parts.length !== 3 && parts.length !== 4) {
+        return [
+          DeeplinksGen.createSetKeybaseLinkError({error}),
+          RouteTreeGen.createNavigateAppend({
+            path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+          }),
+        ]
+      }
+      if (parts[1] === 'new-proof') {
+        return [
+          parts.length === 4 && parts[3] ? ProfileGen.createUpdateUsername({username: parts[3]}) : null,
+          ProfileGen.createAddProof({platform: parts[2], reason: 'appLink'}),
+        ]
+      }
+      break
+    // Fall-through
+    case 'private':
+    case 'public':
+    case 'team':
+      return [
+        RouteTreeGen.createSwitchTab({tab: Tabs.fsTab}),
+        RouteTreeGen.createNavigateAppend({
+          path: [{props: {path: `/keybase/${action.payload.link}`}, selected: 'main'}],
+        }),
+      ]
+    case 'chat':
+      if (parts.length !== 2) {
+        return [
+          DeeplinksGen.createSetKeybaseLinkError({error}),
+          RouteTreeGen.createNavigateAppend({
+            path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+          }),
+        ]
+      }
+      if (parts[1].includes('#')) {
+        const teamChat = parts[1].split('#')
+        if (teamChat.length !== 2) {
+          return [
+            DeeplinksGen.createSetKeybaseLinkError({error}),
+            RouteTreeGen.createNavigateAppend({
+              path: [{props: {errorSource: 'app'}, selected: 'keybaseLinkError'}],
+            }),
+          ]
+        }
+        const [teamname, channelname] = teamChat
+        return [
+          RouteTreeGen.createSwitchTab({tab: Tabs.chatTab}),
+          ChatGen.createPreviewConversation({channelname, reason: 'appLink', teamname}),
+        ]
+      } else {
+        return [
+          RouteTreeGen.createSwitchTab({tab: Tabs.chatTab}),
+          ChatGen.createPreviewConversation({participants: parts[1].split(','), reason: 'appLink'}),
```

Comment:
`keybase://chat/idontexist` navigates to chat with no errors or feedback. maybe we're ok with that, especially as a first pass. looks like it could be tricky to handle those errors here.